code/datums/components/rotation.dm

```diff
@@ -129,41 +119,59 @@
 			rot_degree = 90
 		if(ROTATION_FLIP)
 			rot_degree = 180
-	AM.setDir(turn(AM.dir,rot_degree))
-	after_rotation.Invoke(user,rotation_type)
+	rotated_obj.setDir(turn(rotated_obj.dir, rot_degree))
+	after_rotation.Invoke(user, rotation_type)
 
 /datum/component/simple_rotation/proc/default_can_user_rotate(mob/living/user, rotation_type)
-	if(!istype(user) || !user.canUseTopic(parent, BE_CLOSE, NO_DEXTERITY, FALSE, !iscyborg(user)))
+	if(istype(user) && user.canUseTopic(parent, BE_CLOSE, NO_DEXTERITY, FALSE, !iscyborg(user)))
+		return TRUE
+	if(isobserver(user) && CONFIG_GET(flag/ghost_interaction) && (src.rotation_flags & ROTATION_GHOSTS_ALLOWED))
```

Comment:
```suggestion
	if(isobserver(user) && CONFIG_GET(flag/ghost_interaction) && (rotation_flags & ROTATION_GHOSTS_ALLOWED))
```
Any reason to specify `src` here? There is no local variable with the same name.
Same for the other occurrences.

Reply:
`src` was removed with https://github.com/tgstation/tgstation/pull/64441/commits/e6fc95ffa0352a4f9280d6958a5ada6374a92fbd

I added some variable checks with https://github.com/tgstation/tgstation/pull/64441/commits/2477b39f370d4b27d70e68a70061d64277c7fb12 which basically just returns FALSE.  Not sure if I should toss in some debug messages here or leave it as it is.